### PR TITLE
tcc:  add HEAD mirror

### PR DESCRIPTION
**Pull request by pthariensflame**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This should resolve a current issue where repo.or.cz is down.

### CODE REVIEW

**Comment by pthariensflame** [comment]
So…this doesn't work because `mirror` isn't actually equipped for this.  :/  I naïvely assumed it would have the same capabilities as `url`.

How can we handle this?

File: Formula/tcc.rb [hunk 1]
```diff
@@ -27,6 +26,11 @@ class Tcc < Formula
     sha256 x86_64_linux: "053f79a5752554e18ecba168184e48481bce8a2db418a3f9b0de094f9e6d0e4d"
   end
 
+  head do
+    url "https://repo.or.cz/tinycc.git", branch: "mob"
+    mirror "https://github.com/TinyCC/tinycc.git", branch: "mob"
```
Comment (carlocab) [inline Formula/tcc.rb:31]:
```suggestion
    mirror "https://github.com/TinyCC/tinycc.git"
```
Might work.

File: Formula/tcc.rb [hunk 2]
```diff
@@ -27,6 +26,11 @@ class Tcc < Formula
     sha256 x86_64_linux: "053f79a5752554e18ecba168184e48481bce8a2db418a3f9b0de094f9e6d0e4d"
   end
 
+  head do
+    url "https://repo.or.cz/tinycc.git", branch: "mob"
+    mirror "https://github.com/TinyCC/tinycc.git"
```
Comment (carlocab) [inline Formula/tcc.rb:31]: So this does get rid of the error, but unfortunately `brew` doesn't fall back to the mirror if cloning from the `url` fails.

We could probably just use the GitHub URL for `head` instead.

Comment (carlocab) [reply Formula/tcc.rb:31]: We could maybe add `mirror` support for `head` in `brew`, depending on how complicated it turns out to be. If it's a small change then we'd probably merge a pull request.

Comment (pthariensflame) [reply Formula/tcc.rb:31]: I’ll look into it when I next get the chance.  It seems like it might be useful beyond this formula anyway.

**Comment by carlocab** [comment]
Let's close this for now. We can reopen when we have `mirror` support for `head` in `brew`.

**Comment by pthariensflame** [comment]
Sounds like a plan!

**Pull request closed without merging**